presto-main/src/main/java/com/facebook/presto/catalogserver/CatalogServerClient.java

```diff
@@ -25,32 +25,32 @@
 public interface CatalogServerClient
 {
     @ThriftMethod
-    boolean schemaExists(TransactionInfo transactionInfo, SessionRepresentation session, CatalogSchemaName schema);
+    MetadataEntry<Boolean> schemaExists(TransactionInfo transactionInfo, SessionRepresentation session, CatalogSchemaName schema);
 
     @ThriftMethod
-    boolean catalogExists(TransactionInfo transactionInfo, SessionRepresentation session, String catalogName);
+    MetadataEntry<Boolean> catalogExists(TransactionInfo transactionInfo, SessionRepresentation session, String catalogName);
 
     @ThriftMethod
-    String listSchemaNames(TransactionInfo transactionInfo, SessionRepresentation session, String catalogName);
+    MetadataEntry<String> listSchemaNames(TransactionInfo transactionInfo, SessionRepresentation session, String catalogName);
```

Comment:
I feel this structure might be a bit misleading.
`MetadataEntry<Boolean>` means the result is a `boolean`, while `MetadataEntry<String>` doesn't mean the result is a `String` but a `JSON`. What if the result is a plain string and doesn't need to be converted to `JSON` at all?

Reply:
We can leave it as it is for now. Let's see if other reviewers have better ideas.